webapp/src/components/Metrics/PerformanceAnalysisTable.tsx

```diff
@@ -0,0 +1,646 @@
+      })),
+      {
+        ...DELTA_METRIC_COLUMN,
+        field: "deltaUtteranceCount",
+        headerName: "Delta",
+        cellClassName: "delta",
+        headerAlign: "center",
+        valueGetter: ({ row }) =>
+          row.comparedPipeline &&
+          row.comparedPipeline.utteranceCount - row.basePipeline.utteranceCount,
+        renderCell: ({ value }: GridCellParams<number | undefined>) =>
+          value !== undefined && (
+            <DeltaComputationBar
+              value={value}
+              formattedValue={String(value)}
+              width={Math.abs(value)}
+            />
+          ),
+      },
+      ...ALL_OUTCOMES.flatMap<Column<Row>>((outcome) => [
+        ...pipelines.map<Column<Row>>((pipeline) => ({
+          ...METRIC_COLUMN,
+          field: `${pipeline}${outcome}`,
+          ...groupHeader(
+            pipeline,
+            OutcomeIcon({ outcome }),
+            OUTCOME_PRETTY_NAMES[outcome]
+          ),
+          align: "right",
+          valueGetter: ({ row }) =>
+            row[pipeline] &&
+            row[pipeline]!.outcomeCount[outcome] /
+              row[pipeline]!.utteranceCount,
+          renderCell: ({ value }: GridCellParams<number | undefined>) =>
+            value !== undefined && (
+              <VisualBar
+                formattedValue={formatRatioAsPercentageString(value, 1)}
+                value={value}
+                color={(theme) =>
+                  alpha(theme.palette[OUTCOME_COLOR[outcome]].main, 0.5)
+                }
+              />
+            ),
+        })),
+        {
+          ...DELTA_METRIC_COLUMN,
+          field: `delta${outcome}`,
+          headerName: "Delta",
+          cellClassName: "delta",
+          headerAlign: "center",
+          valueGetter: ({ row }) =>
+            row.comparedPipeline &&
+            row.comparedPipeline.outcomeCount[outcome] /
+              row.comparedPipeline.utteranceCount -
+              row.basePipeline.outcomeCount[outcome] /
+                row.basePipeline.utteranceCount,
+          renderCell: ({ value }: GridCellParams<number | undefined>) =>
+            value !== undefined && (
+              <DeltaComputationBar
+                value={value}
+                formattedValue={formatRatioAsPercentageString(
+                  value as number,
+                  1
+                )}
+                width={isNaN(value) ? 0 : Math.abs(value) * 50}
+              />
+            ),
+        },
+      ]),
+      ...metricsEntries.flatMap<Column<Row>>(
+        ([metricName, { description }]) => [
+          ...pipelines.map<Column<Row>>((pipeline) => ({
+            ...METRIC_COLUMN,
+            field: `${pipeline}${metricName}`,
+            ...groupHeader(pipeline, metricName),
+            description,
+            align: "right",
+            valueGetter: ({ row: { [pipeline]: metrics } }) =>
+              metrics?.customMetrics[metricName] ?? NaN,
```

Comment:
The reason why in https://github.com/ServiceNow/azimuth/pull/228#discussion_r989101289 I suggested
```ts
metrics ? metrics.customMetrics[metricName] ?? NaN : undefined
```
and not
```ts
metrics?.customMetrics[metricName] ?? NaN
```
is because when `metrics` is `undefined`, it means the data is not loaded yet. In that case, we want the cell to be empty, like in the other columns. The only case where we want `NaN` (resulting in `--%`) is when `metrics` is loaded but `metrics.customMetrics` is an empty object (wich makes `metrics.customMetrics[metricName]` undefined).

So, please change
```suggestion
              metrics ? metrics.customMetrics[metricName] ?? NaN : undefined,
```
You could also add a comment if you think it needs clarifying.

Reply:
okay, I added it this way assuming "metrics?" will evaluate if a metric is undefined or not and then fetch custom metrics.